presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -89,6 +92,25 @@
 
     private final SecurityConfig securityConfig;
 
+    /**
+     * Instantiates a new Dispatch Manager.
+     * This is a critical component in resource management section of the query.
+     * This is for the pre-queuing part of central manager for queries
+     *
+     * @param queryIdGenerator the query id generator
+     * @param queryPreparer the query preparer
+     * @param resourceGroupManager the resource group manager
+     * @param warningCollectorFactory the warning collector factory
+     * @param dispatchQueryFactory the dispatch query factory
+     * @param failedDispatchQueryFactory the failed dispatch query factory
+     * @param transactionManager the transaction manager
+     * @param accessControl the access control
+     * @param sessionSupplier the session supplier
+     * @param sessionPropertyDefaults the session property defaults
+     * @param queryManagerConfig the query manager config
+     * @param dispatchExecutor the dispatch executor
+     * @param clusterStatusSender the cluster status sender
```

Comment:
These param are not useful explanation... Maybe just remove them or explain better with why they are needed here.

Same for everything else in the PR

Reply:
yes agree, reworded all params descriptions. I added more description on some key method call, like for how dispatch manager logic flows is covered under `createQuery` session.